### PR TITLE
Issue #515 Adding commit sha to version string

### PR DESCRIPTION
**Pull request by hferentschik**
- Adding test ensuring Minishift vcersion format
- Using Tee in console_test and making sure that Stdout is properly reset
- Clenaing up start_test.go

### CODE REVIEW

**Comment by budhrg** [comment]
@hferentschik , typo if commit description :smile:

**Comment by hferentschik** [comment]
>  typo if commit description

fixed, I hope

**Comment by jorgemoralespou** [comment]
@hferentschik is this now the default output, or only if a -v (or similar is passed)?

**Comment by budhrg** [comment]
> fixed, I hope

Yes @hferentschik

**Comment by hferentschik** [comment]
>  is this now the default output, or only if a -v (or similar is passed)?

default

File: cmd/minishift/cmd/console_test.go [hunk 1]
```diff
@@ -17,42 +17,22 @@ limitations under the License.
...
-	defer os.RemoveAll(testDir)
-
-	f, err := os.Create(testDir + "out.txt")
-	if err != nil {
-		t.Fatal("Error creating test file", err)
-	}
-	defer f.Close()
-
-	os.Stdout = f
+	tee := cli.CreateTee(t, true)
```
Comment (gbraad) [inline cmd/minishift/cmd/console_test.go]: nice...

File: cmd/minishift/cmd/start_test.go [hunk 2]
```diff
@@ -137,13 +135,12 @@ var (
...
 	testRunner *RecordingRunner
+	tee        *cli.Tee
 )
 
 func TestStartClusterUpNoFlags(t *testing.T) {
 	setUp(t)
-	defer os.RemoveAll(testDir)
-	defer minitesting.ResetDefaultRoundTripper()
-	defer viper.Reset()
+	defer tearDown()
```
Comment (gbraad) [inline cmd/minishift/cmd/start_test.go]: +1 removed repetition

File: pkg/version/version.go [hunk 3]
```diff
@@ -27,16 +27,19 @@ const VersionPrefix = "v"
 // The following variables are private fields and should be set when compiling with for example --ldflags="-X github.com/minishift/minishift/pkg/version.openshiftVersion=vX.Y.Z
 var (
 	// The current version of minishift
-	minishiftVersion = "v0.0.0-unset"
+	minishiftVersion = "0.0.0-unset"
```
Comment (gbraad) [inline pkg/version/version.go]: +1

File: Makefile [hunk 4]
```diff
@@ -144,7 +146,7 @@ clean:
 
 .PHONY: test
 test: vendor $(ADDON_ASSET_FILE)
-	@go test -v $(shell $(PACKAGES))
+	@go test -ldflags="$(VERSION_VARIABLES)" -v $(shell $(PACKAGES))
```
Comment (praveenkumar) [inline Makefile]: 👍

**Pull request merged**